### DISCUSSION TITLE
ssh_agent resetting on Ubuntu 20.04 LTS launch

### DISCUSSION THREAD

**Discussion by jaskerv** [microsoft/WSL#5947] [General]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.508]
Your Distribution version: Ubuntu20.04 LTS
Whether the issue is on WSL 2 and/or WSL 1: 2
```

# Steps to reproduce

1. Ensure you have git ssh_agent setup with ssh key linked to git account.
2. Open new Ubuntu Terminal
3. `cd` to git project.
4. perform git action such as `git pull`
5. See that it requires passphrase
`Enter passphrase for key '/home/<user>/.ssh/id_rsa':`
6. Enter passphrase.
7. Repeat steps 2-4. See that it still requires passphrase

#  Expected behavior
shh_agent remembers ssh key has already been setup.

# Actual behavior
ssh_agent resets, requiring the user to re-enter passphrase on new terminal

**Comment by sirredbeard** [comment]
ssh-agent may not be running unless you started it. You will need to manually start it manually, or link to ssh on Windows, there are guides out there for doing each.

**Comment by therealkenc** [comment]
In `.bashrc` something like:
```
export SSH_AGENT_DIR=/tmp/users/$UID
mkdir -p $SSH_AGENT_DIR
if ! pgrep -u "$USER" ssh-agent > /dev/null; then
    ssh-agent > "$SSH_AGENT_DIR/ssh-agent.env"
fi
if [[ ! "$SSH_AUTH_SOCK" ]]; then
    source "$SSH_AGENT_DIR/ssh-agent.env" >/dev/null
fi
```

We'll have a real `$XDG_RUNTIME_DIR` [soon enough](https://github.com/microsoft/WSL/issues/938).

**Comment by therealkenc** [comment]
CoolCold's suggestion over in #3183 ([message](https://github.com/microsoft/WSL/issues/3183)) is pretty good too.
